### ISSUE TITLE
Admin page not redirecting after logging in with Netlify Identity Widget.

### ISSUE THREAD

**Issue by elvisguillen** [konsumer/gatsby-starter-bootstrap-netlify#1]
Great work here! I was able to deploy and access the admin page but after logging in through Netlify it only displays the confirmation window: "Logged in as" but it does not redirect to the actual admin page. I had to refresh the page manually after authorizing to view the admin UI. 

I noticed on the `[Netlify CMS Docs](https://www.netlifycms.org/docs/quick-start/#authentication)` they mentioned placing the redirect script on the main app index page but I'm also seeing it on the admin index file as well on your project, not sure if that is affecting it but I tried swapping it around and ended up breaking the functionality. Let me know if you have any solutions, Thanks!

![netlify-cms-not-redirecting](https://user-images.githubusercontent.com/4971180/32691041-8cf6c384-c6cf-11e7-8ee0-6893eb6ad602.png)

**Comment by konsumer** [comment]
I put that code in the header in the layout, [here](https://github.com/konsumer/gatsby-starter-bootstrap-netlify/blob/master/src/layouts/index.js#L25-L36). I was putting it in the netlify post-processing header (in the netlify build settings), which seemed to work about the same, but this was a way to have it already configured for users.  It seems like the netlify user-stuff is just kinda buggy. I got it working for my blog, sort of, but it still kinda doesn't work sometimes. When it doesn't, I generally have to refresh admin, login, refresh again, etc. Maybe a hacky delay on the redirect would help, not sure. I will play around with the header code and see if I can improve it.

**Comment by konsumer** [comment]
I tried adding a html.js and reworking how stuff is added to the page, and it seems quite a bit more stable. Try it out and let me know how it's working.

**Comment by elvisguillen** [comment]
Just tried it out and the redirect is working great after logging in via Netlify user account, but now I'm seeing "No Entries" message where the posts should be in the Admin UI. Don't think it's a config issue so not sure why, the login definitely seems more stable though.

**Comment by konsumer** [comment]
Nice!

Like I said, I noticed funny little flakey things, but at least for my demo-blog, it seems much better.

I had to make sure a bunch of stuff was setup for it to work:

* Under settings/identity: make sure "Identity service enabled"
* Under settings/identity: I added "External providers" but I dunno if it's needed (I use Github for demo)
* Under settings/identity: Git gateway. I noticed on some sites (like one I made for a friend) I had to turn it off, then back on when I messed with permissions. Also, it's not super-clear in teh docs, but only Github works, for now.
* Under identity: make sure you have your admin user in there. I noticed sometimes I have to remove then re-add a user, it's just kinda flakey

I'm going to close this for now, but it'd be awesome if you reported back with any tricks you used to make it work.
